docs/book/src/reference/using_an_external_type.md

```diff
@@ -0,0 +1,155 @@
+# Using an External Type
+
+There are several different external types that may be referenced when writing a controller.
+* Custom Resource Definitions (CRD) that is defined in the current project `kubebuilder create api`.
+* Core Kubernetes Resources eg. Deployments or Pods.
+* CRD's that are created and installed in another project.
+* A Custom Resource defined via an API Aggregation (AA). Aggregated APIs are subordinate APIServers that sit behind the primary API server, which acts as a proxy.
+
+Currently Kubebuilder handles the first two, CRDs and Core Resources, seamlessly.  External CRDs and CRs created via Aggregation must be scaffolded manually.
+
+In order to use a Kubernetes Custom Resource that has been defined in another project
+you will need to have several items of information.
+* The Domain of the CR
+* The Group under the Domain 
+* The Go import path of the CR Type definition.
+
+The Domain and Group variables have been discussed in other parts of the documentation.  The import path would be located in the project that installs the CR.
+
+This document uses `my` and `their` prefixes as a naming convention for repos, groups, and types to clearly distinguish between your own project and the external one you are referencing.
+
+Example external API Aggregation directory structure
+```
+github.com
+    ├── theiruser
+        ├── theirproject
+            ├── apis
+                ├── theirgroup
+                   ├── doc.go
+                   ├── install
+                   │   ├── install.go
+                   ├── v1alpha1
+                   │   ├── doc.go
+                   │   ├── register.go
+                   │   ├── types.go
+                   │   ├── zz_generated.deepcopy.go
+```
+
+In the case above the import path would be `github.com/theiruser/theirproject/apis/theirgroup/v1alpha1`
+
+### Create a project
+
+```shell
+kubebuilder init --domain $APIDOMAIN --owner "MyCompany"
+```
+
+### Add a controller
+
+be sure to answer no when it asks if you would like to create an api? [Y/n]
+```shell
+kubebuilder create api --group mygroup --version $APIVERSION --kind MyKind
+```
+
+## Edit the API files.
+
```

Comment:
Could we not just only:

Step 1: Get the API module

In your project's directory, run the following command via the command line to get the external type module:
``$ go get -u github.com/external/api``

Step 2: Register the API with the scheme

In the main.go file, add the schema before the line Setup all Controllers:

```go
 ...
    // Adding the external type before setup all Controllers 
	if err := externaltypeimportalias.AddToScheme(mgr.GetScheme()); err != nil {
		log.Error(err, "")
		os.Exit(1)
	}
    
    // Setup all Controllers
	if err := controller.AddToManager(mgr); err != nil {
		log.Error(err, "")
		os.Exit(1)
	}
    ...
```

Reply:
Replaced this with a correct init scheme adjustment in main.go and the test files